Integrations/src/main/java/io/deephaven/integrations/learn/Future.java

```diff
@@ -10,55 +12,55 @@
 
     private final PyObject func;
     private final Input[] inputs;
-    private final IndexSet indexSet;
+    private IndexSet indexSet;
+    private final ColumnSource<?>[][] colSet;
     private boolean called;
     private PyObject result;
 
     /**
      * Creates a new Future.
      *
-     * @param func Function that performs computation on gathered data.
-     * @param inputs Inputs to the Future computation.
-     * @param batchSize Maximum number of rows for deferred computation.
+     * @param func          function that performs computation on gathered data.
+     * @param inputs        inputs to the Future computation.
+     * @param batchSize     maximum number of rows for deferred computation.
      */
-    public Future(PyObject func, int batchSize, Input[] inputs) {
+    Future(PyObject func, int batchSize, Input[] inputs, ColumnSource<?>[][] colSet) {
 
         this.func = func;
         this.inputs = inputs;
         this.indexSet = new IndexSet(batchSize);
+        this.colSet = colSet;
```

Comment:
This looks wrong.  Different inputs need different col sets.  This change has forced them all to one col set, which looks like a problem.  It looks like a bigger problem than the performance problem you were trying to work around.